BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift

```diff
@@ -650,3 +693,114 @@ extension ConvivaAnalytics: UserInterfaceListener {
     }
 }
+
+extension ConvivaAnalytics: SsaiApiDelegate {
+    // swiftlint:disable:next identifier_name
+    var ssaiApi_isAdBreakActive: Bool {
+        isSsaiAdBreakActive
+    }
+
+    func ssaiApi_reportAdBreakStarted() {
+        ssaiApi_reportAdBreakStarted(adBreakInfo: [:])
+    }
+
+    func ssaiApi_reportAdBreakStarted(adBreakInfo: [String: Any]?) {
+        guard !isSsaiAdBreakActive else { return }
+        isSsaiAdBreakActive = true
+
+        videoAnalytics.reportAdBreakStarted(
+            .ADPLAYER_CONTENT,
+            adType: .SERVER_SIDE,
+            adBreakInfo: adBreakInfo ?? [:]
+        )
+    }
+
+    func ssaiApi_reportAdBreakFinished() {
+        guard isSsaiAdBreakActive else { return }
+
+        isSsaiAdBreakActive = false
+        videoAnalytics.reportAdBreakEnded()
+    }
+
+    func ssaiApi_reportAdStarted(adInfo: SsaiAdInfo) {
+        guard isSsaiAdBreakActive else { return }
+
+        adAnalytics.reportAdStarted(adInfo.convivaAdInfo(basedOn: contentMetadataBuilder))
+        adAnalytics.reportAdMetric(CIS_SSDK_PLAYBACK_METRIC_PLAYER_STATE, value: currentPlayerState.rawValue)
+        updateSession()
+    }
+
+    func ssaiApi_reportAdFinished() {
+        guard isSsaiAdBreakActive else { return }
+
+        adAnalytics.reportAdEnded()
+    }
+
+    func ssaiApi_reportAdSkipped() {
+        guard isSsaiAdBreakActive else { return }
+
+        adAnalytics.reportAdSkipped()
+    }
+
+    func ssaiApi_update(adInfo: SsaiAdInfo) {
+        guard isSsaiAdBreakActive else { return }
+
+        adAnalytics.setAdInfo(adInfo.convivaAdInfo(basedOn: contentMetadataBuilder))
+    }
+}
+
+private extension SsaiAdInfo {
+    func convivaAdInfo(basedOn contentMetadataBuilder: ContentMetadataBuilder) -> [String: Any] {
+        var adInfo = [String: Any]()
+        adInfo["c3.ad.id"] = notAvailable
+        adInfo["c3.ad.system"] = notAvailable
+        adInfo["c3.ad.mediaFileApiFramework"] = notAvailable
+        adInfo["c3.ad.firstAdSystem"] = notAvailable
+        adInfo["c3.ad.firstAdId"] = notAvailable
+        adInfo["c3.ad.firstCreativeId"] = notAvailable
+        adInfo["c3.ad.technology"] = "Server Side"
+
+        adInfo["c3.ad.isSlate"] = isSlate ? "true" : "false"
+
+        if let title {
+            adInfo[CIS_SSDK_METADATA_ASSET_NAME] = title
+        }
+        if let duration {
+            adInfo[CIS_SSDK_METADATA_DURATION] = duration
+        }
+        if let id {
+            adInfo["c3.ad.id"] = id
+        }
+        if let adSystem {
+            adInfo["c3.ad.system"] = adSystem
+        }
+        if let position {
+            adInfo["c3.ad.position"] = position.convivaAdPosition.rawValue
+        }
+        if let adStitcher {
+            adInfo["c3.ad.stitcher"] = adStitcher
+        }
+
+        additionalMetadata?.forEach { key, value in
+            adInfo[key] = value
+        }
+
+        let mergedAdInfo = contentMetadataBuilder
+            .build()
+            .merging(adInfo) { $1 }
```

Comment:
TBD if we want to carry over the already collected metadata.

This will take adInfo with priority over content metadata when values exists for both.